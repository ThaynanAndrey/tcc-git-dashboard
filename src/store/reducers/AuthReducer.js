import { USER_LOGIN_SUCCESS, USER_LOGIN_ERROR, USER_LOGOUT_SUCCESS, USER_LOGOUT_ERROR } from '../actions/types';

export const initState = {
    user: undefined,
    accessToken: undefined,
    errorLogin: undefined,
    errorLogout: undefined
};
  
const authReducer = (state = initState, action) => {
    switch(action.type) {
        case USER_LOGIN_SUCCESS:
            return {
                ...state,
                user: action.user,
                idUser: action.idUser,
                accessToken: action.accessToken
            };
        case USER_LOGIN_ERROR:
            return {
                ...state,
                errorLogin: action.errorLogin
            }
        case USER_LOGOUT_SUCCESS:
            return initState;
        case USER_LOGOUT_ERROR:
            return {
                ...state,
                errorLogout: action.errorLogout
            }
        default:
          return state
    }
};
  
export default authReducer;