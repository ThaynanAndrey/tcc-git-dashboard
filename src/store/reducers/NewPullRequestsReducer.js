import { LOAD_PULL_REQUESTS_NO_PROJECT_SUCCESS, ADDED_PULL_REQUEST_SUCCESS, ADDED_PULL_REQUEST_ERROR } from '../actions/types';

const initState = {
    pullRequestsNoProject: []
};
  
const newPullRequestsReducer = (state = initState, action) => {
    switch(action.type) {
        case LOAD_PULL_REQUESTS_NO_PROJECT_SUCCESS:
            console.log('pull requests no project loaded!');
            return {
                ...state,
                pullRequestsNoProject: action.pullRequestsNoProject
            };
        case ADDED_PULL_REQUEST_SUCCESS:
            console.log("Pull request added");
            return state;
        case ADDED_PULL_REQUEST_ERROR:
            console.log("Error to add pull request");
            return state;
        default:
          return state
    }
};
  
export default newPullRequestsReducer;