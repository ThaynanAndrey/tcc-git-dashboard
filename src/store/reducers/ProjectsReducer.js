import { LOAD_PROJECTS_SUCCESS, LOAD_PROJECTS_ERROR, CREATE_PROJECT_SUCCESS,
    CREATE_PROJECT_ERROR, GET_PROJECT_SUCCESS } from '../actions/types';

export const initState = {
    projects: [],
    project: undefined,
    currentProject: undefined,
    error: undefined
};
  
const projectsReducer = (state = initState, action) => {
    switch(action.type) {
        case LOAD_PROJECTS_SUCCESS:
            return {
                ...state,
                projects: action.projects
            };
        case LOAD_PROJECTS_ERROR:
            return {
                ...state,
                projects: [],
                error: action.error
            };
        case CREATE_PROJECT_SUCCESS:
            return {
                ...state,
                projects: action.projects
            };
        case GET_PROJECT_SUCCESS:
            return {
                ...state,
                project: action.project
            }
        case CREATE_PROJECT_ERROR:
            return state;
        default:
          return state;
    }
};
  
export default projectsReducer;