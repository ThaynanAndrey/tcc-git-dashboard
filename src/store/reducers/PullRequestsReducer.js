import { LOAD_PROJECT_PULL_REQUESTS_SUCCESS, LOAD_PULL_REQUEST_SUCCESS, RESET_PROJECT_PULL_REQUESTS } from '../actions/types';

const initState = {
    projectPullRequests: [],
    selectedPullRequest: { responsavel: {}, repositorio: {} },
};
  
const pullRequestsReducer = (state = initState, action) => {
    switch(action.type) {
        case LOAD_PROJECT_PULL_REQUESTS_SUCCESS:
            console.log('get all pull requests from project!');
            return {
                ...state,
                projectPullRequests: action.projectPullRequests
            };
        case LOAD_PULL_REQUEST_SUCCESS:
            return {
                ...state,
                selectedPullRequest: action.selectedPullRequest
            }
        case RESET_PROJECT_PULL_REQUESTS:
            return {
                ...state,
                projectPullRequests: []
            }
        default:
          return state
    }
};
  
export default pullRequestsReducer;