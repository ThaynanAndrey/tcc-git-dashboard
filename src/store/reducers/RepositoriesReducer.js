import { LOAD_PROJECT_REPOSITORIES_SUCCESS, LOAD_REPOSITORIES_NO_PROJECT_SUCCESS, ADDED_REPOSITORY_SUCCESS,
    ADDED_REPOSITORY_ERROR, LOAD_EXTERNAL_REPOSITORIES, LOAD_EXTERNAL_REPOSITORIES_ERROR, RESET_PROJECT_REPOSITORIES,
    RESET_REPOSITORIES_NO_PROJECT } from '../actions/types';

export const initState = {
    projectRepositories: [],
    repositoriesNoProject: [],
    externalRepositories: [],
    msgExternalRepositories: undefined,
    error: undefined
};
  
const repositoriesReducer = (state = initState, action) => {
    switch(action.type) {
        case LOAD_PROJECT_REPOSITORIES_SUCCESS:
            return {
                ...state,
                projectRepositories: action.projectRepositories
            };
        case LOAD_REPOSITORIES_NO_PROJECT_SUCCESS:
            return {
                ...state,
                repositoriesNoProject: action.repositoriesNoProject
            }
        case ADDED_REPOSITORY_SUCCESS:
            return {
                ...state,
                repositoriesNoProject: action.repositoriesNoProject,
                externalRepositories: action.externalRepositories
            }
        case ADDED_REPOSITORY_ERROR:
            return {
                ...state,
                error: action.error
            }
        case LOAD_EXTERNAL_REPOSITORIES:
        case LOAD_EXTERNAL_REPOSITORIES_ERROR:
            return {
                ...state,
                externalRepositories: action.externalRepositories,
                msgExternalRepositories: action.msgExternalRepositories
            }
        case RESET_PROJECT_REPOSITORIES:
            return {
                ...state,
                projectRepositories: []
            }
        case RESET_REPOSITORIES_NO_PROJECT:
            return {
                ...state,
                repositoriesNoProject: [],
                externalRepositories: []
            }
        default:
          return state
    }
};
  
export default repositoriesReducer;