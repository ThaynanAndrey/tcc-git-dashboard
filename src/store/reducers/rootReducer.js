import { combineReducers } from 'redux';
import authReducer from './AuthReducer';
import pullRequestsReducer from './PullRequestsReducer';
import newPullRequestsReducer from './NewPullRequestsReducer';
import repositoriesReducer from './RepositoriesReducer';

const rootReducer = combineReducers({
    auth: authReducer,
    pullRequests: pullRequestsReducer,
    newPullRequests: newPullRequestsReducer,
    repositories: repositoriesReducer
});

export default rootReducer;