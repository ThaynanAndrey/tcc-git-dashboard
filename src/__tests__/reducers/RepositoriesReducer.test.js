import repositoriesReducer from '../../store/reducers/RepositoriesReducer';
import { initState } from '../../store/reducers/RepositoriesReducer';

describe('RepositoriesReducer', () => {
    it('should return the initial state', () => {
        const state = repositoriesReducer(undefined, {});
        
        expect(state).toMatchSnapshot();
        expect(state).toEqual(initState);
    });

    it('should handle LOAD_PROJECT_REPOSITORIES_SUCCESS', () => {
        const projectRepositories = [
            { name: "My repo", date: "10/11/2012" },
            { name: "My second repo", date: "11/11/2012" }
        ];
        
        const action = {
            type: 'LOAD_PROJECT_REPOSITORIES_SUCCESS',
            projectRepositories
        };
        
        const newState = {
            ...initState,
            projectRepositories
        };

        const state = repositoriesReducer(undefined, action);

        expect(state).toMatchSnapshot();
        expect(state).toEqual(newState);
    });

    it('should handle LOAD_REPOSITORIES_NO_PROJECT_SUCCESS', () => {
        const repositoriesNoProject = [
            { name: "out repo, not mine", date: "10/11/2012" }
        ];
        
        const action = {
            type: 'LOAD_REPOSITORIES_NO_PROJECT_SUCCESS',
            repositoriesNoProject
        };
        
        const newState = {
            ...initState,
            repositoriesNoProject
        };

        const state = repositoriesReducer(undefined, action);

        expect(state).toMatchSnapshot();
        expect(state).toEqual(newState);
    });

    it('should handle ADDED_REPOSITORY_SUCCESS', () => {
        const repositoriesNoProject = [
            { name: "My repo", date: "10/11/2012" },
            { name: "My second repo", date: "11/11/2012" }
        ];
        const externalRepositories = [
            { name: "My external repo", date: "13/11/2012" }
        ];
        
        const action = {
            type: 'ADDED_REPOSITORY_SUCCESS',
            repositoriesNoProject,
            externalRepositories
        };
        
        const newState = {
            ...initState,
            repositoriesNoProject,
            externalRepositories
        };

        const state = repositoriesReducer(undefined, action);

        expect(state).toMatchSnapshot();
        expect(state).toEqual(newState);
    });

    it('should handle ADDED_REPOSITORY_ERROR', () => {
        const action = {
            type: 'ADDED_REPOSITORY_ERROR',
            error: 'Shiiiii'
        };
        
        const newState = {
            ...initState,
            error: 'Shiiiii'
        };

        const state = repositoriesReducer(undefined, action);

        expect(state).toMatchSnapshot();
        expect(state).toEqual(newState);
    });

    it('should handle LOAD_EXTERNAL_REPOSITORIES', () => {
        const externalRepositories = [
            { name: "My repo", date: "10/11/2012" },
            { name: "My second repo", date: "11/11/2012" }
        ];
        const msgExternalRepositories = "";
        
        const action = {
            type: 'LOAD_EXTERNAL_REPOSITORIES',
            externalRepositories,
            msgExternalRepositories
        };
        
        const newState = {
            ...initState,
            externalRepositories,
            msgExternalRepositories
        };

        const state = repositoriesReducer(undefined, action);

        expect(state).toMatchSnapshot();
        expect(state).toEqual(newState);
    });

    it('should handle LOAD_EXTERNAL_REPOSITORIES_ERROR', () => {
        const externalRepositories = [
            { name: "My repo", date: "10/11/2012" },
            { name: "My second repo", date: "11/11/2012" }
        ];
        const msgExternalRepositories = "Oppss, its wrong...";
        
        const action = {
            type: 'LOAD_EXTERNAL_REPOSITORIES_ERROR',
            externalRepositories,
            msgExternalRepositories
        };
        
        const newState = {
            ...initState,
            externalRepositories,
            msgExternalRepositories
        };

        const state = repositoriesReducer(undefined, action);

        expect(state).toMatchSnapshot();
        expect(state).toEqual(newState);
    });

    it('should handle RESET_PROJECT_REPOSITORIES', () => {
        const action = {
            type: 'RESET_PROJECT_REPOSITORIES'
        };
        
        const newState = {
            ...initState,
            projectRepositories: []
        };

        const state = repositoriesReducer(undefined, action);

        expect(state).toMatchSnapshot();
        expect(state).toEqual(newState);
    });

    it('should handle RESET_REPOSITORIES_NO_PROJECT', () => {
        const action = {
            type: 'RESET_REPOSITORIES_NO_PROJECT'
        };
        
        const newState = {
            ...initState,
            repositoriesNoProject: [],
            externalRepositories: []
        };

        const state = repositoriesReducer(undefined, action);

        expect(state).toMatchSnapshot();
        expect(state).toEqual(newState);
    });
});