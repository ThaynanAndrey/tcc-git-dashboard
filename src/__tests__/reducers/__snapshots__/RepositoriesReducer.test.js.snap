// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepositoriesReducer should handle ADDED_REPOSITORY_ERROR 1`] = `
Object {
  "error": "Shiiiii",
  "externalRepositories": Array [],
  "msgExternalRepositories": undefined,
  "projectRepositories": Array [],
  "repositoriesNoProject": Array [],
}
`;

exports[`RepositoriesReducer should handle ADDED_REPOSITORY_SUCCESS 1`] = `
Object {
  "error": undefined,
  "externalRepositories": Array [
    Object {
      "date": "13/11/2012",
      "name": "My external repo",
    },
  ],
  "msgExternalRepositories": undefined,
  "projectRepositories": Array [],
  "repositoriesNoProject": Array [
    Object {
      "date": "10/11/2012",
      "name": "My repo",
    },
    Object {
      "date": "11/11/2012",
      "name": "My second repo",
    },
  ],
}
`;

exports[`RepositoriesReducer should handle LOAD_EXTERNAL_REPOSITORIES 1`] = `
Object {
  "error": undefined,
  "externalRepositories": Array [
    Object {
      "date": "10/11/2012",
      "name": "My repo",
    },
    Object {
      "date": "11/11/2012",
      "name": "My second repo",
    },
  ],
  "msgExternalRepositories": "",
  "projectRepositories": Array [],
  "repositoriesNoProject": Array [],
}
`;

exports[`RepositoriesReducer should handle LOAD_EXTERNAL_REPOSITORIES_ERROR 1`] = `
Object {
  "error": undefined,
  "externalRepositories": Array [
    Object {
      "date": "10/11/2012",
      "name": "My repo",
    },
    Object {
      "date": "11/11/2012",
      "name": "My second repo",
    },
  ],
  "msgExternalRepositories": "Oppss, its wrong...",
  "projectRepositories": Array [],
  "repositoriesNoProject": Array [],
}
`;

exports[`RepositoriesReducer should handle LOAD_PROJECT_REPOSITORIES_SUCCESS 1`] = `
Object {
  "error": undefined,
  "externalRepositories": Array [],
  "msgExternalRepositories": undefined,
  "projectRepositories": Array [
    Object {
      "date": "10/11/2012",
      "name": "My repo",
    },
    Object {
      "date": "11/11/2012",
      "name": "My second repo",
    },
  ],
  "repositoriesNoProject": Array [],
}
`;

exports[`RepositoriesReducer should handle LOAD_REPOSITORIES_NO_PROJECT_SUCCESS 1`] = `
Object {
  "error": undefined,
  "externalRepositories": Array [],
  "msgExternalRepositories": undefined,
  "projectRepositories": Array [],
  "repositoriesNoProject": Array [
    Object {
      "date": "10/11/2012",
      "name": "out repo, not mine",
    },
  ],
}
`;

exports[`RepositoriesReducer should handle RESET_PROJECT_REPOSITORIES 1`] = `
Object {
  "error": undefined,
  "externalRepositories": Array [],
  "msgExternalRepositories": undefined,
  "projectRepositories": Array [],
  "repositoriesNoProject": Array [],
}
`;

exports[`RepositoriesReducer should handle RESET_REPOSITORIES_NO_PROJECT 1`] = `
Object {
  "error": undefined,
  "externalRepositories": Array [],
  "msgExternalRepositories": undefined,
  "projectRepositories": Array [],
  "repositoriesNoProject": Array [],
}
`;

exports[`RepositoriesReducer should return the initial state 1`] = `
Object {
  "error": undefined,
  "externalRepositories": Array [],
  "msgExternalRepositories": undefined,
  "projectRepositories": Array [],
  "repositoriesNoProject": Array [],
}
`;
